fedimint-api/src/db/mod.rs

```diff
@@ -680,6 +680,46 @@ where
     }
 }
 
+/// This is a helper macro that generates the implementations of
+/// DatabaseKeyPrefixConst necessary for reading/writing to the
+/// database and fetching by prefix.
+///
+/// - `$key`: This is the type of struct that will be used as the key into the database.
+/// - `$val`: This is the type of struct that will be used as the value into the database.
+/// - `$prefix`: Enum expression that is represented as a `u8` that is the prefix assigned to this key.
+/// - `$prefix_ty`: Optional type of struct that will be used as the key prefix when querying the database via `find_by_prefix`.
+///
+/// Examples:
+///
+/// `impl_db_prefix_const!(TestKey, TestVal, TestDbKeyPrefix::Test);`
+///
+/// `impl_db_prefix_const!(AltTestKey, TestVal, TestDbKeyPrefix::AltTest, AltDbPrefixTestPrefix);`
+#[macro_export]
+macro_rules! impl_db_prefix_const {
+    ($key:ty, $val:ty, $prefix:expr) => {
+        impl DatabaseKeyPrefixConst for $key {
+            const DB_PREFIX: u8 = $prefix as u8;
+            type Key = Self;
+            type Value = $val;
+        }
+    };
+    ($key:ty, $val:ty, $prefix:expr, $($prefix_ty:ty),+) => {
```

Comment:
any reason for multiple rules?

use named arguments in macros whenever the order can be not obvious. 
```suggestion
    (key = $key:ty, value = $val:ty, prefix = $prefix:expr $(, key_prefix = $prefix_ty:ty)* $(,)?) => {
```